src/box/lua/call.c

```diff
@@ -1268,6 +1269,46 @@ lbox_func_new_or_delete(struct trigger *trigger, void *event)
 	return 0;
 }
 
+static int
+lbox_rt_access_reset(struct lua_State *L)
+{
+	if (lua_gettop(L) != 0)
+		return luaL_error(L, "internal.reset_lua_call takes no "
+				  "arguments, but %d given.", lua_gettop(L));
+
+	struct credentials *cr = effective_user();
+	if (cr->uid != ADMIN)
+		return luaL_error(L, "internal.reset_lua_call only ADMIN "
+				  "can grant access.");
```

Comment:
I believe this check is necessary for the following reasons. When using `require('net.box').connect`, even if the user has the privileges to execute Lua functions `lua_call: [all]`, this does not grant them privileges to execute built-in functions. In this case, `box.session.su` would be inaccessible.

Reply:
> I believe this check is necessary for the following reasons. When using `require('net.box').connect`, even if the user has the privileges to execute Lua functions `lua_call: [all]`, this does not grant them privileges to execute built-in functions. In this case, `box.session.su` would be inaccessible.

These functions should be added to the built-in function list (along with all other box functions) so granting `lua_call: [all]` wouldn't grant the user access to it.

> In addition, @Totktonada and I have decided that this API should only be used by the `admin`, and granting other users the rights to use this API should not be allowed.

We don't do access checks for other internal functions. I don't see why these internal functions should be different.

Reply:
The `lua_call: [all]` point actually doesn't work: it doesn't give an access to these internal functions, because they're in the `box` namespace. Also, I see no other ways to grant access to call these functions accidentally without granting everything.

It seems, there are no strict point for and against the check. They all about consistency.

We can look on the consistency here from two points of view.

1. It is a Lua code and it doesn't usually perform access checks. This is consistency with other functions implemented in Lua.
2. It has the same effect as granting `lua_call` using `box.schema.user.grant()` or a direct write to `_priv` -- the call becomes allowed. These actions are [forbidden](https://github.com/tarantool/tarantool/pull/10358#discussion_r1728984204) for all except the admin using a low level code -- the `_priv` trigger in `alter.cc`. It seems consistent that actions that have the same effect are allowed for the same group of users.

I'm for consistency and I'm not really insist on consistency of the second kind. I think that the effect is more important than how it is implemented, but I can find a logic in the first type of consistency too.

If @locker insist on the first type of consistency, OK.

Reply:
Since everything will work just fine even if we don't add this access check, I think we shouldn't add it.

Reply:
Check for `ADMIN` dropped.